File: app/src/main/java/com/anokmik/databinding/communicator/LoginFragmentCommunicator.java
package com.anokmik.databinding.communicator;

import android.content.Context;
import android.databinding.ObservableBoolean;
import android.databinding.adapters.TextViewBindingAdapter;
import android.text.TextUtils;
import android.widget.Toast;

import com.anokmik.databinding.R;

public class LoginFragmentCommunicator {

    private final Context context;

    private String loginValue;

    public ObservableBoolean isLoginValid = new ObservableBoolean(false);

    public ObservableBoolean isPasswordValid = new ObservableBoolean(false);

    public LoginFragmentCommunicator(Context context) {
        this.context = context;
    }

    public String getLoginValue() {
        return loginValue;
    }

    public void setLoginValue(String loginValue) {
        this.loginValue = loginValue;
    }

    public TextViewBindingAdapter.OnTextChanged onLoginTextChanged = new TextViewBindingAdapter.OnTextChanged() {
        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {
            isLoginValid.set(!TextUtils.isEmpty(s));
        }
    };

    public TextViewBindingAdapter.OnTextChanged onPasswordTextChanged = new TextViewBindingAdapter.OnTextChanged() {
        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {
            isPasswordValid.set(isLoginValid.get() & (s.length() >= 6));
        }
    };

    public void showToast() {
        Toast.makeText(context, context.getResources()
                .getString(R.string.greeting_sign_in, loginValue), Toast.LENGTH_SHORT).show();
    }

}

File: app/src/main/kotlin/com/anokmik/databinding/communicator/LoginFragmentCommunicator.kt
package com.anokmik.databinding.communicator;

import android.content.Context
import android.databinding.ObservableBoolean
import android.databinding.adapters.TextViewBindingAdapter
import android.text.TextUtils
import android.widget.Toast
import com.anokmik.databinding.R

class LoginFragmentCommunicator(val context: Context) {

    var loginValue: String? = null

    @JvmField
    val isLoginValid: ObservableBoolean = ObservableBoolean(false)

    @JvmField
    val isPasswordValid: ObservableBoolean = ObservableBoolean(false)

    @JvmField
    val onLoginTextChanged: TextViewBindingAdapter.OnTextChanged = TextViewBindingAdapter.OnTextChanged { s, start, before, count -> isLoginValid.set(!TextUtils.isEmpty(s)) }

    @JvmField
    val onPasswordTextChanged: TextViewBindingAdapter.OnTextChanged = TextViewBindingAdapter.OnTextChanged { s, start, before, count -> isPasswordValid.set(isLoginValid.get() && (s.length >= 6)) }

    fun showToast() {
        Toast.makeText(context, context.resources.getString(R.string.greeting_sign_in, loginValue), Toast.LENGTH_SHORT).show()
    }

}